abdiff/core/collate_ab_transforms.py

```diff
@@ -85,22 +110,44 @@ def get_transformed_records_iter(
 
     * timdex_record_id: The TIMDEX record ID.
     * source: The shorthand name of the source as denoted in by Transmogrifier
-      (see https://github.com/MITLibraries/transmogrifier/blob/main/transmogrifier/config.py).
+    * run_date: Run date from TIMDEX ETL
+    * run_type: "full" or "daily"
+    * action: "index" or "delete"
     * record: The TIMDEX record serialized to a JSON string then encoded to bytes.
     * version: The version of the transform, parsed from the absolute filepath to a
       transformed file.
     * transformed_file_name: The name of the transformed file, excluding file extension.
     """
     version = get_transform_version(transformed_file)
     filename_details = parse_timdex_filename(transformed_file)
-    with open(transformed_file, "rb") as file:
-        for record in ijson.items(file, "item"):
+
+    base_record = {
+        "source": filename_details["source"],
+        "run_date": filename_details["run-date"],
+        "run_type": filename_details["run-type"],
+        "action": filename_details["action"],
+        "version": version,
+        "transformed_file_name": transformed_file.split("/")[-1],
+    }
+
+    # handle JSON files with records to index
+    if transformed_file.endswith(".json"):
+        with open(transformed_file, "rb") as file:
+            for record in ijson.items(file, "item"):
+                yield {
+                    **base_record,
+                    "timdex_record_id": record["timdex_record_id"],
+                    "record": json.dumps(record).encode(),
+                }
+
+    # handle TXT files with records to delete
+    else:
+        deleted_records_df = pd.read_csv(transformed_file, header=None)
+        for row in deleted_records_df.itertuples():
             yield {
-                "timdex_record_id": record["timdex_record_id"],
-                "source": filename_details["source"],
-                "record": json.dumps(record).encode(),
-                "version": version,
-                "transformed_file_name": transformed_file.split("/")[-1],
+                **base_record,
+                "timdex_record_id": row[1],
+                "record": None,
```

Comment:
This is `None`, because we don't care about a deleted record's actual record body!  If it's the _last_ instance of that record in the run, then it will be removed entirely.  Otherwise, the more recent version -- which will have a record -- will be utilized.